pkgs/by-name/ni/nix-update/package.nix

```
@@ -1,42 +1,62 @@
-{ lib
-, python3
-, fetchFromGitHub
-, nix
-, nix-prefetch-git
-, nixpkgs-review
+{
+  lib,
+  callPackage,
+  fetchFromGitHub,
+  nix,
+  nix-prefetch-git,
+  nixpkgs-review,
+  python3Packages,
 }:
 
-python3.pkgs.buildPythonApplication rec {
-  pname = "nix-update";
-  version = "1.5.1";
-  pyproject = true;
+let
+  self = python3Packages.buildPythonApplication {
+    pname = "nix-update";
+    version = "1.5.1";
+    pyproject = true;
 
-  src = fetchFromGitHub {
-    owner = "Mic92";
-    repo = pname;
-    rev = "refs/tags/${version}";
-    hash = "sha256-JXls4EgMDAWtd736nXS2lYTUv9QIjRpkCTimxNtMN7Q=";
-  };
+    src = fetchFromGitHub {
+      owner = "Mic92";
+      repo = "nix-update";
+      rev = "refs/tags/${self.version}";
+      hash = "sha256-JXls4EgMDAWtd736nXS2lYTUv9QIjRpkCTimxNtMN7Q=";
+    };
+
+    nativeBuildInputs = [ python3Packages.setuptools ];
+
+    makeWrapperArgs = [
+      "--prefix"
+      "PATH"
+      ":"
+      (lib.makeBinPath [
+        nix
+        nix-prefetch-git
+        nixpkgs-review
+      ])
+    ];
+
+    installCheckPhase = ''
```

```suggestion
    checkPhase = ''
```
is the same for Python packages

---

It is a bit strange since it calls installed programs. But OK then.